mm2src/lp_ordermatch.rs

```diff
@@ -3540,6 +3550,27 @@ pub async fn order_status(ctx: MmArc, req: Json) -> Result<Response<Vec<u8>>, St
             .map_err(|e| ERRL!("{}", e));
     }
 
+    let order_path = my_orders_history_dir(&ctx).join(req.uuid.to_string() + ".json");
+
+    if let Ok(order) = json::from_slice::<MakerOrder>(&slurp(&order_path)) {
+        let res = json!({
+            "type": "Maker",
```

Comment:
Adding the tag to the Order enum and saving it to history it will allow avoiding this and the code will be much shorter.